addons/wiki/models.py

```diff
@@ -82,6 +85,241 @@ def build_wiki_url(node, label, base, end):
+        contributors = [user._id for user in node.contributors]
+        wiki_utils.broadcast_to_sharejs('reload',
+                                        sharejs_uuid,
+                                        data=contributors)
+
+    def belongs_to_node(self, node_id):
+        """Check whether the wiki is attached to the specified node."""
+        return self.node._id == node_id
+
+    @property
+    def current_version_number(self):
+        if self.versions.exists():
+            return self.versions.count()
+        return 0
+
+    @property
+    def url(self):
+        return u'{}wiki/{}/'.format(self.node.url, self.page_name)
+
+    def create_version(self, user, content, created=None, modified=None, date=None):
+        if created and modified and date:
+            version = WikiVersion(user=user, wiki_page=self, content=content, identifier=self.current_version_number + 1, created=created, modified=modified, date=date)
+        else:
+            version = WikiVersion(user=user, wiki_page=self, content=content, identifier=self.current_version_number + 1)
+        version.save()
+        return version
+
+    def get_version(self, version=None, required=False):
+        if version is None:
+            if self.versions.exists():
+                return self.versions.last()
+            return None
+        try:
+            return self.versions.get(identifier=version)
+        except WikiVersion.DoesNotExist:
+            if required:
+                raise VersionNotFoundError(version)
+            return None
+
+    def get_versions(self):
+        return self.versions.all().order_by('-date')
+
+    def rename(self, new_name, save=True):
+        self.page_name = new_name
+        if save:
+            self.save()
+
+    @property
+    def root_target_page(self):
+        """The comment page type associated with WikiPages."""
+        return 'wiki'
+
+    @property
+    def deep_url(self):
+        return u'{}wiki/{}/'.format(self.node.deep_url, self.page_name)
+
+    def clone_wiki(self, node_id):
+        """Clone a wiki page.
+        :param node: The Node of the cloned wiki page
+        :return: The cloned wiki page
+        """
+        node = AbstractNode.load(node_id)
+        if not node:
+            raise ValueError('Invalid node')
+        clone = self.clone()
+        clone.node = node
+        clone.user = self.user
+        clone.save()
+        return clone
+
+    @classmethod
+    def clone_wiki_versions(cls, node, copy, user, save=True):
+        """Clone wiki pages for a forked or registered project.
+        First clones the WikiPage, then clones all WikiPage versions.
+        :param node: The Node that was forked/registered
+        :param copy: The fork/registration
+        :param user: The user who forked or registered the node
+        :param save: Whether to save the fork/registration
+        :return: copy
+        """
+        for wiki_page in node.wikis.all():
```

Comment:
Should we only clone non-deleted pages? I *think* from the previous logic deleted pages wouldn't have been cloned.